src/query-builder/select-query-builder.ts

```diff
@@ -1414,6 +1415,87 @@ export interface SelectQueryBuilder<DB, TB extends keyof DB, O>
    */
   $castTo<T>(): SelectQueryBuilder<DB, TB, T>
 
+  /**
+   * Changes the output type from an object to a tuple.
+   *
+   * This doesn't affect the generated SQL in any way. This function is
+   * just a necessary evil when you need to convert a query's output
+   * record type to a tuple type. Typescript doesn't currently offer
+   * tools to do this automatically (without insane hackery).
+   *
+   * The returned object can no longer be executed. It can only be used
+   * as a subquery.
+   *
+   * ### Examples
+   *
+   * ```ts
+   * const result = await db
+   *   .selectFrom('person')
+   *   .selectAll('person')
+   *   .where(({ eb, refTuple, selectFrom }) => eb(
+   *     refTuple('first_name', 'last_name'),
+   *     'in',
+   *     selectFrom('pet')
+   *       .select(['name', 'species'])
+   *       .where('pet.species', '!=', 'cat')
+   *       .$asTuple('name', 'species')
+   *   ))
+   * ```
+   *
+   * The generated SQL(PostgreSQL):
+   *
+   * ```sql
+   * select
+   *   "person".*
+   * from
+   *   "person"
+   * where
+   *   ("first_name", "last_name")
+   *   in
+   *   (
+   *     select "name", "species"
+   *     from "pet"
+   *     where "pet"."species" != $1
+   *   )
+   * ```
+   */
+  $asTuple<K1 extends keyof O, K2 extends keyof O>(
+    key1: K1,
+    key2: K2
+  ): ExpressionWrapper<DB, TB, [O[K1], O[K2]]>
+
+  $asTuple<K1 extends keyof O, K2 extends keyof O, K3 extends keyof O>(
+    key1: K1,
+    key2: K2,
+    key3: K3
+  ): ExpressionWrapper<DB, TB, [O[K1], O[K2], O[K3]]>
+
+  $asTuple<
+    K1 extends keyof O,
+    K2 extends keyof O,
+    K3 extends keyof O,
+    K4 extends keyof O
+  >(
+    key1: K1,
+    key2: K2,
+    key3: K3,
+    key4: K4
+  ): ExpressionWrapper<DB, TB, [O[K1], O[K2], O[K3], O[K4]]>
+
+  $asTuple<
+    K1 extends keyof O,
+    K2 extends keyof O,
+    K3 extends keyof O,
+    K4 extends keyof O,
+    K5 extends keyof O
+  >(
+    key1: K1,
+    key2: K2,
+    key3: K3,
+    key4: K4,
+    key5: K5
+  ): ExpressionWrapper<DB, TB, [O[K1], O[K2], O[K3], O[K4], O[K5]]>
```

Comment:
Is it possible to enforce there are no duplicates here?

Something like:

```suggestion
  $asTuple<K1 extends keyof O, K2 extends Exclude<keyof O, K1>>(
    key1: K1,
    key2: K2
  ): ExpressionWrapper<DB, TB, [O[K1], O[K2]]>

  $asTuple<
    K1 extends keyof O,
    K2 extends Exclude<keyof O, K1>,
    K3 extends Exclude<keyof O, K1 | K2>
  >(
    key1: K1,
    key2: K2,
    key3: K3
  ): ExpressionWrapper<DB, TB, [O[K1], O[K2], O[K3]]>

  $asTuple<
    K1 extends keyof O,
    K2 extends Exclude<keyof O, K1>,
    K3 extends Exclude<keyof O, K1 | K2>,
    K4 extends Exclude<keyof O, K1 | K2 | K3>
  >(
    key1: K1,
    key2: K2,
    key3: K3,
    key4: K4
  ): ExpressionWrapper<DB, TB, [O[K1], O[K2], O[K3], O[K4]]>

  $asTuple<
    K1 extends keyof O,
    K2 extends Exclude<keyof O, K1>,
    K3 extends Exclude<keyof O, K1 | K2>,
    K4 extends Exclude<keyof O, K1 | K2 | K3>,
    K5 extends Exclude<keyof O, K1 | K2 | K3 | K4>
  >(
    key1: K1,
    key2: K2,
    key3: K3,
    key4: K4,
    key5: K5
  ): ExpressionWrapper<DB, TB, [O[K1], O[K2], O[K3], O[K4], O[K5]]>
```

Reply:
Also, there's probably a way to enforce all keys from `O` were used.

Reply:
That just enforces one of the columns.

Reply:
No wait... You're right. That should work